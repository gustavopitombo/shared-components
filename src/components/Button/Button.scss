button.shd-button {
    background: none;
    background-color: #1976d2;
    border: 0;
    border-radius: 4px;
    border-style: none;
    box-sizing: border-box;
    cursor: pointer;
    color: #fff;
    font-family: inherit;
    font-size: 100%; 
    line-height: 1.15; 
    margin: 0;
    outline: 0;
    overflow: visible;
    padding: 6px 16px;
    text-transform: none;
    user-select: none;
    -webkit-appearance: button;
}