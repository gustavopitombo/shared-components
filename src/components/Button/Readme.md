Basic button:

```tsx
<Button onClick={() => alert("Hi!")}>Push me!</Button>
```