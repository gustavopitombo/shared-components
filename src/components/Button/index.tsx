import React, { ReactNode } from 'react';

type ButtonProps = {
    children: ReactNode,
    onClick?: () => void,
}

export const Button = ({ children, onClick }: ButtonProps) => (
    <button data-testid="shd-button" className="shd-button" onClick={onClick}>
        {children}
    </button>
)

export default Button;
