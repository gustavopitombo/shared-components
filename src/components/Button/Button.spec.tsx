import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Button from './index';

describe("Button component", () => {
    it("render button", () => {
        const { getByTestId } = render(<Button>Clique aqui</Button>);   
        expect(getByTestId("shd-button")).toHaveTextContent("Clique aqui");
    });

    it("matches snapshot", () => {
        const tree = renderer.create(<Button>Clique aqui</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})