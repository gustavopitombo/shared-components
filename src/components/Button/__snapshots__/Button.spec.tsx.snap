// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  className="shd-button"
  data-testid="shd-button"
>
  Clique aqui
</button>
`;
